### PR TITLE
Upgrade dropwizard and other dependencies.

### LINKED ISSUE #126: Upgrade Dropwizard version
While this project lay dormant on Dropwizard 0.8, Dropwizard reached 1.0. Upgrade to latest.

### PR DESCRIPTION
**Pull request by jshort**
Given that active development on coner is about to startup again, I went
ahead and updated a lot of other dependencies.  I also made a lot of
checkstyle rules 'error' since we were already compliant and only marked
the few that we don't care about as 'warning'.

There were some changes necessary to get the DAO unit tests to work but
when the new DAOTestRule:

https://github.com/dropwizard/dropwizard/blob/master/dropwizard-testing/src/main/java/io/dropwizard/testing/junit/DAOTestRule.java

Is integrated into a release of dropwizard-testing, we can make use of
it in a very clean way.

Resolves #126

### CODE REVIEW

**Comment by jshort** [comment]
Note that I ran the coner server and did some simple GET/POST on /events to make sure nothing grossly regressed.

File: pom.xml [hunk 1]
```diff
@@ -35,11 +35,11 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-checkstyle-plugin</artifactId>
-                <version>2.13</version>
+                <version>2.17</version>
                 <configuration>
                     <configLocation>build-tools/src/main/resources/checkstyle/checkstyle.xml</configLocation>
                     <includeTestSourceDirectory>true</includeTestSourceDirectory>
-                    <suppressionsLocation>checkstyle/suppressions.xml</suppressionsLocation>
+                    <!-- <suppressionsLocation>checkstyle/suppressions.xml</suppressionsLocation> -->
```
Comment (carltonwhitehead) [inline pom.xml]: Why did you comment this out?

File: dropwizard-app/pom.xml [hunk 2]
```diff
@@ -42,9 +42,30 @@
         </dependency>
 
         <dependency>
+            <groupId>com.smoketurner</groupId>
+            <artifactId>dropwizard-swagger</artifactId>
+            <version>1.0.5-3</version>
```
Comment (carltonwhitehead) [inline dropwizard-app/pom.xml]: I started up the server using `java -jar dropwizard-app/target/dropwizard-app-0.1-SNAPSHOT.jar server dropwizard-app/src/test/resources/config/test.yml`, but when I request http://localhost:8080/swagger.json, I get an HTTP 500 and the app prints a stack trace pointing to https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java#L47, which suggests something stopped swagger obtaining resource paths.

Comment (carltonwhitehead) [inline dropwizard-app/pom.xml]: Expected response body:

```json
{
    "basePath": "/", 
    "definitions": {
        "AddCompetitionGroupRequest": {
            "properties": {
                "grouping": {
                    "default": false, 
                    "type": "boolean"
                }, 
                "handicapFactor": {
                    "format": "double", 
                    "maximum": 1.0, 
                    "minimum": 0.0, 
                    "type": "number"
                }, 
                "name": {
                    "type": "string"
                }, 
                "resultTimeType": {
                    "type": "string"
                }
            }, 
            "required": [
                "grouping", 
                "handicapFactor"
            ]
        }, 
        "AddCompetitionGroupSetRequest": {
            "properties": {
                "competitionGroups": {
                    "items": {
                        "$ref": "#/definitions/CompetitionGroup"
                    }, 
                    "type": "array", 
                    "uniqueItems": true
                }, 
                "name": {
                    "type": "string"
                }
            }
        }, 
        "AddEventRequest": {
            "properties": {
                "date": {
                    "format": "date-time", 
                    "type": "string"
                }, 
                "name": {
                    "type": "string"
                }
            }, 
            "required": [
                "date"
            ]
        }, 
        "AddHandicapGroupRequest": {
            "properties": {
                "handicapFactor": {
                    "format": "double", 
                    "maximum": 1.0, 
                    "minimum": 0.0, 
                    "type": "number"
                }, 
                "name": {
                    "type": "string"
                }
            }, 
            "required": [
                "handicapFactor"
            ]
        }, 
        "AddHandicapGroupSetRequest": {
            "properties": {
                "handicapGroups": {
                    "items": {
                        "$ref": "#/definitions/HandicapGroup"
                    }, 
                    "type": "array", 
                    "uniqueItems": true
                }, 
                "name": {
                    "type": "string"
                }
            }
        }, 
        "AddRegistrationRequest": {
            "properties": {
                "firstName": {
                    "type": "string"
                }, 
                "lastName": {
                    "type": "string"
                }
            }
        }, 
        "CompetitionGroup": {
            "properties": {
                "id": {
                    "type": "string"
                }
            }
        }, 
        "CompetitionGroupApiEntity": {
            "properties": {
                "grouping": {
                    "default": false, 
                    "type": "boolean"
                }, 
                "handicapFactor": {
                    "format": "double", 
                    "maximum": 1.0, 
                    "minimum": 0.0, 
                    "type": "number"
                }, 
                "id": {
                    "type": "string"
                }, 
                "name": {
                    "type": "string"
                }, 
                "resultTimeType": {
                    "type": "string"
                }
            }, 
            "required": [
                "grouping", 
                "handicapFactor"
            ]
        }, 
        "CompetitionGroupSetApiEntity": {
            "properties": {
                "competitionGroups": {
                    "items": {
                        "$ref": "#/definitions/CompetitionGroupApiEntity"
                    }, 
                    "type": "array", 
                    "uniqueItems": true
                }, 
                "id": {
                    "type": "string"
                }, 
                "name": {
                    "type": "string"
                }
            }
        }, 
        "ErrorsResponse": {
            "properties": {
                "errors": {
                    "items": {
                        "type": "string"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "EventApiEntity": {
            "properties": {
                "date": {
                    "format": "date-time", 
                    "type": "string"
                }, 
                "id": {
                    "type": "string"
                }, 
                "name": {
                    "type": "string"
                }
            }, 
            "required": [
                "date"
            ]
        }, 
        "GetCompetitionGroupSetsResponse": {
            "properties": {
                "competitionGroupSets": {
                    "items": {
                        "$ref": "#/definitions/CompetitionGroupSetApiEntity"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "GetCompetitionGroupsResponse": {
            "properties": {
                "competitionGroups": {
                    "items": {
                        "$ref": "#/definitions/CompetitionGroupApiEntity"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "GetEventRegistrationsResponse": {
            "properties": {
                "registrations": {
                    "items": {
                        "$ref": "#/definitions/RegistrationApiEntity"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "GetEventsResponse": {
            "properties": {
                "events": {
                    "items": {
                        "$ref": "#/definitions/EventApiEntity"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "GetHandicapGroupsResponse": {
            "properties": {
                "handicapGroups": {
                    "items": {
                        "$ref": "#/definitions/HandicapGroupApiEntity"
                    }, 
                    "type": "array"
                }
            }
        }, 
        "HandicapGroup": {
            "properties": {
                "id": {
                    "type": "string"
                }
            }
        }, 
        "HandicapGroupApiEntity": {
            "properties": {
                "handicapFactor": {
                    "format": "double", 
                    "maximum": 1.0, 
                    "minimum": 0.0, 
                    "type": "number"
                }, 
                "id": {
                    "type": "string"
                }, 
                "name": {
                    "type": "string"
                }
            }, 
            "required": [
                "handicapFactor"
            ]
        }, 
        "RegistrationApiEntity": {
            "properties": {
                "firstName": {
                    "type": "string"
                }, 
                "id": {
                    "type": "string"
                }, 
                "lastName": {
                    "type": "string"
                }
            }, 
            "required": [
                "firstName"
            ]
        }
    }, 
    "info": {}, 
    "paths": {
        "/competitionGroups": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getCompetitionGroups", 
                "parameters": [], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "successful operation", 
                        "schema": {
                            "$ref": "#/definitions/GetCompetitionGroupsResponse"
                        }
                    }
                }, 
                "summary": "Get all Competition Groups", 
                "tags": [
                    "Competition Groups"
                ]
            }, 
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "addCompetitionGroup", 
                "parameters": [
                    {
                        "description": "Competition Group", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddCompetitionGroupRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created at URI in Location header"
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add a Competition Group", 
                "tags": [
                    "Competition Groups"
                ]
            }
        }, 
        "/competitionGroups/sets": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getCompetitionGroupSets", 
                "parameters": [], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "successful operation", 
                        "schema": {
                            "$ref": "#/definitions/GetCompetitionGroupSetsResponse"
                        }
                    }
                }, 
                "summary": "Get all Competition Group Sets", 
                "tags": [
                    "Competition Groups"
                ]
            }, 
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "Optionally include a list of Competition Group entities with ID to associate them", 
                "operationId": "add", 
                "parameters": [
                    {
                        "description": "Competition Group Set", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddCompetitionGroupSetRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created", 
                        "headers": {}
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add a new Competition Group Set", 
                "tags": [
                    "Competition Groups"
                ]
            }
        }, 
        "/competitionGroups/sets/{competitionGroupSetId}": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getCompetitionGroupSet", 
                "parameters": [
                    {
                        "description": "Competition Group Set ID", 
                        "in": "path", 
                        "name": "competitionGroupSetId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "OK", 
                        "schema": {
                            "$ref": "#/definitions/CompetitionGroupApiEntity"
                        }
                    }, 
                    "404": {
                        "description": "Not found", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get a Competition Group Set", 
                "tags": [
                    "Competition Groups"
                ]
            }
        }, 
        "/competitionGroups/{competitionGroupId}": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getCompetitionGroup", 
                "parameters": [
                    {
                        "description": "Competition Group ID", 
                        "in": "path", 
                        "name": "competitionGroupId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "OK", 
                        "schema": {
                            "$ref": "#/definitions/CompetitionGroupApiEntity"
                        }
                    }, 
                    "404": {
                        "description": "Not found", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get a Competition Group", 
                "tags": [
                    "Competition Groups"
                ]
            }
        }, 
        "/events": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getEvents", 
                "parameters": [], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "successful operation", 
                        "schema": {
                            "$ref": "#/definitions/GetEventsResponse"
                        }
                    }
                }, 
                "summary": "Get a list of all events", 
                "tags": [
                    "Events"
                ]
            }, 
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "addEvent", 
                "parameters": [
                    {
                        "description": "Event", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddEventRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created at URI in Location header"
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add an Event", 
                "tags": [
                    "Events"
                ]
            }
        }, 
        "/events/{eventId}": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getEvent", 
                "parameters": [
                    {
                        "description": "Event ID", 
                        "in": "path", 
                        "name": "eventId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "OK", 
                        "schema": {
                            "$ref": "#/definitions/EventApiEntity"
                        }
                    }, 
                    "404": {
                        "description": "Not found", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get an Event", 
                "tags": [
                    "Events"
                ]
            }
        }, 
        "/events/{eventId}/registrations": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getEventRegistrations", 
                "parameters": [
                    {
                        "description": "Event ID", 
                        "in": "path", 
                        "name": "eventId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "Success", 
                        "schema": {
                            "$ref": "#/definitions/GetEventRegistrationsResponse"
                        }
                    }, 
                    "404": {
                        "description": "No event with given ID", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get a list of all registrations at an event", 
                "tags": [
                    "Event Registrations"
                ]
            }, 
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "addRegistration", 
                "parameters": [
                    {
                        "description": "Event ID", 
                        "in": "path", 
                        "name": "eventId", 
                        "required": true, 
                        "type": "string"
                    }, 
                    {
                        "description": "Registration", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddRegistrationRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created at URI in Location header"
                    }, 
                    "404": {
                        "description": "No event with given ID", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add a new registration", 
                "tags": [
                    "Event Registrations"
                ]
            }
        }, 
        "/events/{eventId}/registrations/{registrationId}": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getRegistration", 
                "parameters": [
                    {
                        "description": "Event ID", 
                        "in": "path", 
                        "name": "eventId", 
                        "required": true, 
                        "type": "string"
                    }, 
                    {
                        "description": "Registration ID", 
                        "in": "path", 
                        "name": "registrationId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "OK", 
                        "schema": {
                            "$ref": "#/definitions/RegistrationApiEntity"
                        }
                    }, 
                    "404": {
                        "description": "Not found", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }, 
                    "409": {
                        "description": "Event ID and Registration ID are mismatched", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get a specific registration", 
                "tags": [
                    "Event Registrations"
                ]
            }
        }, 
        "/handicapGroups": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getHandicapGroups", 
                "parameters": [], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "successful operation", 
                        "schema": {
                            "$ref": "#/definitions/GetHandicapGroupsResponse"
                        }
                    }
                }, 
                "summary": "Get all Handicap Groups", 
                "tags": [
                    "Handicap Groups"
                ]
            }, 
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "addHandicapGroup", 
                "parameters": [
                    {
                        "description": "Handicap Group", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddHandicapGroupRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created at URI in Location header"
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add a Handicap Group", 
                "tags": [
                    "Handicap Groups"
                ]
            }
        }, 
        "/handicapGroups/sets": {
            "post": {
                "consumes": [
                    "application/json"
                ], 
                "description": "Optionally include a set of Handicap Group entities with ID to associate them", 
                "operationId": "add", 
                "parameters": [
                    {
                        "description": "Handicap Group Set", 
                        "in": "body", 
                        "name": "body", 
                        "required": false, 
                        "schema": {
                            "$ref": "#/definitions/AddHandicapGroupSetRequest"
                        }
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "201": {
                        "description": "Created", 
                        "headers": {}
                    }, 
                    "422": {
                        "description": "Failed validation", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Add a new Handicap Group Set", 
                "tags": [
                    "Handicap Groups"
                ]
            }
        }, 
        "/handicapGroups/{handicapGroupId}": {
            "get": {
                "consumes": [
                    "application/json"
                ], 
                "description": "", 
                "operationId": "getHandicapGroup", 
                "parameters": [
                    {
                        "description": "Handicap Group ID", 
                        "in": "path", 
                        "name": "handicapGroupId", 
                        "required": true, 
                        "type": "string"
                    }
                ], 
                "produces": [
                    "application/json"
                ], 
                "responses": {
                    "200": {
                        "description": "OK", 
                        "schema": {
                            "$ref": "#/definitions/HandicapGroupApiEntity"
                        }
                    }, 
                    "404": {
                        "description": "Not found", 
                        "schema": {
                            "$ref": "#/definitions/ErrorsResponse"
                        }
                    }
                }, 
                "summary": "Get a Handicap Group", 
                "tags": [
                    "Handicap Groups"
                ]
            }
        }
    }, 
    "swagger": "2.0", 
    "tags": [
        {
            "name": "Events"
        }, 
        {
            "name": "Handicap Groups"
        }, 
        {
            "name": "Event Registrations"
        }, 
        {
            "name": "Competition Groups"
        }
    ]
}
```

Comment (jshort) [inline dropwizard-app/pom.xml]: Before my commit:

    GET     /swagger.json (com.wordnik.swagger.jaxrs.listing.ApiListingResource)
    GET     /swagger.yaml (com.wordnik.swagger.jaxrs.listing.ApiListingResource)

With my commit:

    GET     /swagger.{type:json|yaml} (io.swagger.jaxrs.listing.ApiListingResource)

Comment (jshort) [inline dropwizard-app/pom.xml]: Figure it out.  The com.wordnik stuff was moved do io.swagger and revved the version to match that of swagger-core.

The old dropwizard-swagger dependency brought a lot of transitive dependencies that had the planets align to just work.  Now we need to manage them ;).

File: pom.xml (hunk 1 above)
Comment (jshort) [inline pom.xml]: Just a test and even with the new strict, build failing checkstyle config, the suppressions are not necessary, but I'll leave them if we want to add to them in the future.

**Pull request merged**